### PR TITLE
Spanner float32 support

### PR DESCRIPTION
**Pull request by shubhamswe**
Fixes #<issue_number_goes_here>

**[DO NOT REVIEW]** This PR is work in progress. 

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: sources/postgres/toddl_test.go [hunk 1]
```diff
@@ -253,7 +251,7 @@ func TestToExperimentalSpannerType(t *testing.T) {
 		ColIds: []string{"c1", "c2", "c3", "c4", "c5", "c6", "c7"},
 		ColDefs: map[string]ddl.ColumnDef{
 			"c1": ddl.ColumnDef{Name: "a", Id: "c1", T: ddl.Type{Name: ddl.Int64}},
-			"c2": ddl.ColumnDef{Name: "b", Id: "c2", T: ddl.Type{Name: ddl.Float64}},
+			"c2": ddl.ColumnDef{Name: "b", Id: "c2", T: ddl.Type{Name: ddl.Float32}},
```
Comment (arawind) [inline sources/postgres/toddl_test.go:254]: Please add a test for FLOAT64 too

File: sources/mysql/toddl_test.go [hunk 2]
```diff
@@ -275,7 +274,7 @@ func TestToSpannerPostgreSQLDialectType(t *testing.T) {
 		ColIds: []string{"c1", "c2", "c3", "c4", "c5", "c6", "c7", "c8", "c9", "c10"},
 		ColDefs: map[string]ddl.ColumnDef{
 			"c1":  ddl.ColumnDef{Name: "a", Id: "c1", T: ddl.Type{Name: ddl.Int64}},
-			"c2":  ddl.ColumnDef{Name: "b", Id: "c2", T: ddl.Type{Name: ddl.Float64}},
+			"c2":  ddl.ColumnDef{Name: "b", Id: "c2", T: ddl.Type{Name: ddl.Float32}},
```
Comment (arawind) [inline sources/mysql/toddl_test.go:277]: Same here, please add a FLOAT64 column too

File: sources/mysql/mysqldump_test.go [hunk 3]
```diff
@@ -819,12 +819,12 @@ func TestProcessMySQLDump_DataError(t *testing.T) {
 			expectedData: []spannerData{
 				spannerData{
 					table: "test", cols: []string{"a", "b", "c", "d", "e", "f", "g", "synth_id"},
-					vals: []interface{}{int64(7), float64(42.1), true,
+					vals: []interface{}{int64(7), float32(42.1), true,
```
Comment (arawind) [inline sources/mysql/mysqldump_test.go:822]: Please add a double column and set a float64 value too

File: sources/postgres/toddl_test.go (hunk 1 above)
Comment (shubhamswe) [reply sources/postgres/toddl_test.go]: done

File: sources/mysql/toddl_test.go (hunk 2 above)
Comment (shubhamswe) [reply sources/mysql/toddl_test.go:279]: done.

File: sources/mysql/mysqldump_test.go (hunk 3 above)
Comment (shubhamswe) [reply sources/mysql/mysqldump_test.go]: done.

**Comment by manitgupta** [comment]
@shreyakhajanchi Please review.

File: sources/csv/data.go [hunk 4]
```diff
@@ -379,6 +379,24 @@ func convArray(spannerType ddl.Type, val string) (interface{}, error) {
 			r = append(r, spanner.NullDate{Date: date, Valid: true})
 		}
 		return r, nil
+	case ddl.Float32:
```
Comment (shreyakhajanchi) [inline sources/csv/data.go:382]: Please add a UT for this

File: sources/csv/toddl.go [hunk 5]
```diff
@@ -36,6 +36,8 @@ func ToSpannerType(columnType string) (ddl.Type, error) {
 		return ddl.Type{}, fmt.Errorf("%v is not a valid Spanner column type", columnType)
 	case ty == "DATE":
 		return ddl.Type{Name: ddl.Date}, nil
+	case ty == "FLOAT32":
```
Comment (shreyakhajanchi) [inline sources/csv/toddl.go:39]: Please add a UT

File: sources/mysql/toddl.go [hunk 6]
```diff
@@ -85,7 +85,7 @@ func toSpannerTypeInternal(srcType schema.Type, spType string) (ddl.Type, []inte
 		case ddl.String:
```
Comment (shreyakhajanchi) [inline sources/mysql/toddl.go:85]: Is the FLOAT32 datatype being added for PG dialect as well? If yes, what is the DDL for it and if no then that handling is missing from the code throughout.

**Review by shreyakhajanchi — COMMENTED** [review]
Is float32 being added to PG dialect as well ? If the syntax differs or if its not added then PG dialect handling for Float32 should also be added

File: sources/csv/data.go (hunk 4 above)
Comment (shubhamswe) [reply sources/csv/data.go:382]: done.

File: sources/csv/toddl.go (hunk 5 above)
Comment (shubhamswe) [reply sources/csv/toddl.go:39]: done.

File: sources/mysql/toddl.go (hunk 6 above)
Comment (shubhamswe) [reply sources/mysql/toddl.go:85]: Yes, FLOAT32 datatype is added for PG dialect and is represented by "float4" or "real" keywords.

Comment (shreyakhajanchi) [reply sources/mysql/toddl.go:85]: In that case, it needs to be added in STANDARD_TYPE_TO_PGSQL_TYPEMAP and PGSQL_TO_STANDARD_TYPE_TYPEMAP in ast.go and test cases for it in ast_test.go

Comment (shubhamswe) [reply sources/mysql/toddl.go:85]: done.

**Pull request closed without merging**